controllers/sensor.go

```diff
@@ -0,0 +1,122 @@
+package controllers
```

Comment:
This seems pretty nice tbh. Straight to the point, it's obvious what's happening. But you are no longer using this? The logic for getting sensor information is obvious and I don't have to dig through multiple files to see what is going on. Having a single file that handles your logic is not a bad thing.